verifier/src/quote_enclave_identity.rs

```diff
@@ -116,11 +160,106 @@ impl<'a> TryFrom<&'a str> for SignedQeIdentity<'a> {
     }
 }
 
+/// Verifier for ensuring a QE(Quoting Enclave) identity was signed with the
+/// provided key
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SignedQeIdentityVerifier {
+    key: VerifyingKey,
+    time: DateTime,
+}
+
+impl SignedQeIdentityVerifier {
+    /// Create a new instance.
+    ///
+    /// The `SignedQeIdentityVerifier::verify()` will fail if the signature doesn't
+    /// match for the `key` or if the `time` is outside the `issueDate` and
+    /// `nextUpdate` times in the `enclaveIdentity`.
+    ///
+    /// # Arguments
+    /// - `key` - The public key to verify the `enclaveIdentity` signature with
+    ///   This should be retrieved from the x509 certificate provided in the
+    ///   QE identity request from
+    ///   <https://api.trustedservices.intel.com/sgx/certification/v4/qe/identity?update=standard>
+    /// - `time` - The current system time
+    ///   This is expected to be generated by:
+    ///     ```ignore
+    ///     let time = DateTime::from_system_time(SystemTime::now()).unwrap();
+    ///     ```
+    ///   or equivalent
+    pub fn new(key: VerifyingKey, time: DateTime) -> Self {
+        Self { key, time }
+    }
+}
+
+impl<'a, E: Accessor<SignedQeIdentity<'a>>> Verifier<E> for SignedQeIdentityVerifier {
+    type Value = Option<Error>;
+    fn verify(&self, evidence: &E) -> VerificationOutput<Self::Value> {
+        let signed_qe_identity = evidence.get();
+        let result = signed_qe_identity.verify(&self.key, self.time);
+        let is_success = result.is_ok() as u8;
+
+        let mut error = None;
+        if let Err(e) = result {
+            error = Some(e);
+        }
+
+        VerificationOutput::new(error, is_success.into())
+    }
+}
+
+impl VerificationMessage<Option<Error>> for SignedQeIdentityVerifier {
```

Comment:
IFF the SignedJson approach is desired, I think only these verification messages would need to be implemented for each type.